### PR TITLE
Feat/schema-overhaul

### PR DESCRIPTION
**Pull request by hugo-nl**
- Obs: This PR is not ready for review, but if you're curios you can look at the schema.prisma file to see how we might want to restructure our database schema
- In this schema we take fully advantages of the relational db where we can use cascading effects
- In order to do this we have to move the fk in lots of places so it the correct child entity cascades if the parent is deleted
- Still lots some stuff to cleanup in the routes but I think this will enable us to offer a more robust and maintainable api 
- Bonus: This will enable us to to use null for garboAI since we can migrate to using actual deletes instead of sending in null for cruD operations

TLDR:
New Schema design --> Hopefully more sustainable :)

### CODE REVIEW

**Review by Greenheart — COMMENTED** [review]
This looks really promising! Indeed, this will enable several improvements going forward.

File: src/lib/prisma.ts [hunk 1]
```diff
@@ -660,6 +680,25 @@ export async function upsertReportingPeriod(
...
+        },
+      },
+      economy: {
+        include: {
+          employees: { select: { id: true } },
+          turnover: { select: { id: true } },
+        },
+      },
+      company: { select: { wikidataId: true } },
+    },
```
[the remark this answers is not in the document]
Comment (hugo-nl) [reply src/lib/prisma.ts:701]: Most definitely! Not ready for review :D

File: prisma/schema.prisma [hunk 2]
```diff
@@ -135,86 +126,78 @@ model Emissions {
...
-  metadata  Metadata   @relation(fields: [metadataId], references: [id])
+  emissions Emissions? @relation(fields: [emissionsId], references: [id], onDelete: Cascade)
+  metadata  Metadata[]
 }
 
 /// Biogenic emissions are reported separately from scope 1-3
-/// If we want to save a more detailed breakdown (when companies reported this), we can use the metadata comment to save this context
 model BiogenicEmissions {
-  id         Int    @id @default(autoincrement())
-  /// Sometimes companies break it down into scope 1-3 - however these should always be stored as a total number according to the GHG protocol.
```
[the remark this answers is not in the document]
Comment (hugo-nl) [reply prisma/schema.prisma:164]: Agreed! Bringing that one back 🙌
